monitors/SyntheticPegMonitor.js

```diff
@@ -0,0 +1,100 @@
+const { createFormatFunction, createEtherscanLinkMarkdown } = require("../common/FormattingUtils");
+
+class SyntheticPegMonitor {
+  /**
+   * @notice Constructs new Liquidator bot.
+   * @param {Object} logger Module used to send logs.
+   * @param {Object} uniswapPriceFeed Module used to query the current uniswap token price.
+   * @param {Object} cryptoWatchPriceFeed Module used to query the current crypto watch token price.
+   * @param {Object} [config] Contains fields with which constructor will attempt to override defaults.
+   */
+  constructor(logger, uniswapPriceFeed, cryptoWatchPriceFeed) {
+    this.logger = logger;
+
+    // Instance of price feeds used to check for deviation of synthetic token price.
+    this.uniswapPriceFeed = uniswapPriceFeed;
+    this.cryptoWatchPriceFeed = cryptoWatchPriceFeed;
+
+    this.web3 = this.uniswapPriceFeed.web3;
+
+    // Contract constants
+    // TODO: replace this with an actual query to the collateral currency symbol
+    this.collateralCurrencySymbol = "DAI";
+    this.syntheticCurrencySymbol = "UMATEST";
+
+    // TODO: get the decimals of the collateral currency and use this to scale the output appropriately for non 1e18 colat
+    this.formatDecimalString = createFormatFunction(this.web3, 2);
+
+    // If the price of one price feed deviates by more than `deviationAlertThreshold` percent from the other, trigger alert.
+    this.deviationAlertThreshold = this.web3.utils.toBN(this.web3.utils.toWei("20"));
+
+    // Default config settings. SyntheticPegMonitor deployer can override these settings by passing in new
+    // values via the `config` input object. The `isValid` property is a function that should be called
+    // before resetting any config settings. `isValid` must return a Boolean.
+    const defaultConfig = {
+      deviationAlertThreshold: {
+        // `deviationAlertThreshold`: Percentage error threshold used to compare observed and expected token prices.
+        // if the deviation in token price exceeds this value an alert is fired.
+        value: 20,
+        isValid: x => {
+          return x >= 0 && x < 100;
+        }
+      }
+    };
+  }
+
+  // Queries disputable liquidations and disputes any that were incorrectly liquidated.
+  checkPriceDeviation = async () => {
+    this.logger.debug({
+      at: "SyntheticPegMonitor",
+      message: "Checking price deviation"
+    });
+    // Get the latest prices from the two price feeds.
+    const uniswapTokenPrice = this.uniswapPriceFeed.getLastBlockPrice();
+    const cryptoWatchTokenPrice = this.cryptoWatchPriceFeed.getCurrentPrice();
```

Comment:
Note that this does not use the twap method. This was considered but it makes more sense to take the most recent price as the current value.